### PR TITLE
Add documentation for wildcard field type

### PR DESCRIPTION
**Pull request by msfroh**
### Description
Adds documentation for the `wildcard` field type added in https://github.com/opensearch-project/OpenSearch/pull/13461

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._
N/A

### Version
_List the OpenSearch version to which this PR applies, e.g. 2.14, 2.12--2.14, or all._
Tentatively 2.15, depending on whether https://github.com/opensearch-project/OpenSearch/pull/13461 gets merged and backported.

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by msfroh** [comment]
Woohoo! The code was merged. So, this will make it into 2.15.

**Review by kolchfa-aws — APPROVED** [review]
Thank you, @msfroh! Could you review the example that I added on line 16 and then we'll move the PR to editorial review.

File: _field-types/supported-field-types/wildcard.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+grand_parent: Supported field types
+---
+
+# Wildcard field type
+
+A `wildcard` field is a variant of a `keyword` field designed for arbitrary substring and regular expression matching.
+
+Use a `wildcard` field when your content consists of "strings of characters" and not "text". Examples include unstructured log lines and computer code.
+
+The `wildcard` field type is indexed differently from the `keyword` field type. Whereas `keyword` fields write the original field value to the index, the `wildcard` field type splits the field value into substrings with length less than or equal to 3 and writes the substrings to the index. For example, the string `test` is split into strings `t`, `te`, `tes`, `e`, `es`, and `est`. At search time, required substrings from the query pattern are matched against the index to produce candidate documents, which are then filtered according to the pattern in the query.
```
Comment (natebower) [inline _field-types/supported-field-types/wildcard.md:16]:
```suggestion
The `wildcard` field type is indexed differently from the `keyword` field type. Whereas `keyword` fields write the original field value to the index, the `wildcard` field type splits the field value into substrings with a length that is less than or equal to 3 and writes the substrings to the index. For example, the string `test` is split into strings `t`, `te`, `tes`, `e`, `es`, and `est`. At search time, required substrings from the query pattern are matched against the index to produce candidate documents, which are then filtered according to the pattern in the query.
```

File: _field-types/supported-field-types/wildcard.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+In general, exact match queries (like [`term`]({{site.url}}{{site.baseurl}}/query-dsl/term/term/) or [`terms`]({{site.url}}{{site.baseurl}}/query-dsl/term/term/) queries) perform worse on `wildcard` fields than on `keyword` fields, while [`wildcard`]({{site.url}}{{site.baseurl}}/query-dsl/term/wildcard/), [`prefix`]({{site.url}}{{site.baseurl}}/query-dsl/term/prefix/), and [`regexp`]({{site.url}}{{site.baseurl}}/query-dsl/term/regexp/) queries perform better on `wildcard` fields.
```
Comment (natebower) [inline _field-types/supported-field-types/wildcard.md:18]:
```suggestion
In general, exact match queries (like [`term`]({{site.url}}{{site.baseurl}}/query-dsl/term/term/) or [`terms`]({{site.url}}{{site.baseurl}}/query-dsl/term/term/) queries) perform less effectively on `wildcard` fields than on `keyword` fields, while [`wildcard`]({{site.url}}{{site.baseurl}}/query-dsl/term/wildcard/), [`prefix`]({{site.url}}{{site.baseurl}}/query-dsl/term/prefix/), and [`regexp`]({{site.url}}{{site.baseurl}}/query-dsl/term/regexp/) queries perform better on `wildcard` fields.
```

File: _field-types/supported-field-types/wildcard.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+## Parameters
+
+The following table lists all parameters available for `wildcard` fields.
+
+Parameter | Description
+:--- | :---
+`doc_values` | A Boolean value that specifies whether the field should be stored on disk so that it can be used for aggregations, sorting, or scripting. Default is `false`.
+`ignore_above` | Any string longer than this integer value should not be indexed. Default is `2147483647`.
+`normalizer` | The normalizer used to preprocess values for indexing and search. By default, no normalization occurs and the original value is used. You may use the `lowercase` normalizer to perform case-insentive matching on the field.
+`null_value` | A value to be used in place of `null`. Must be of the same type as the field. If this parameter is not specified, the field is treated as missing when its value is `null`. Default is `null`.
```
Comment (natebower) [inline _field-types/supported-field-types/wildcard.md:48]:
```suggestion
`null_value` | A value to be used in place of `null`. Must be of the same type as the field. If this parameter is not specified, then the field is treated as missing when its value is `null`. Default is `null`.
```

**Review by natebower — COMMENTED** [review]
@msfroh @kolchfa-aws Just a few changes. Thanks!

File: _field-types/supported-field-types/wildcard.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+grand_parent: Supported field types
+---
+
+# Wildcard field type
+
+A `wildcard` field is a variant of a `keyword` field designed for arbitrary substring and regular expression matching.
+
+Use a `wildcard` field when your content consists of "strings of characters" and not "text". Examples include unstructured log lines and computer code.
+
+The `wildcard` field type is indexed differently from the `keyword` field type. Whereas `keyword` fields write the original field value to the index, the `wildcard` field type splits the field value into substrings with a length that is less than or equal to 3 and writes the substrings to the index. For example, the string `test` is split into strings `t`, `te`, `tes`, `e`, `es`, and `est`. At search time, required substrings from the query pattern are matched against the index to produce candidate documents, which are then filtered according to the pattern in the query.
```
Comment (msfroh) [inline _field-types/supported-field-types/wildcard.md:16]: Specifically, at search time, if we query for the term `test` we do an indexed search for `tes AND est`. (We only use the 1-2 character substrings when the search term is shorter than 3 characters.)

For each matching document, we check if the source value is `test` (to exclude false positive values like `nikola tesla felt alternating current was best`).

File: _field-types/supported-field-types/wildcard.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+In general, exact match queries (like [`term`]({{site.url}}{{site.baseurl}}/query-dsl/term/term/) or [`terms`]({{site.url}}{{site.baseurl}}/query-dsl/term/term/) queries) perform less effectively on `wildcard` fields than on `keyword` fields, while [`wildcard`]({{site.url}}{{site.baseurl}}/query-dsl/term/wildcard/), [`prefix`]({{site.url}}{{site.baseurl}}/query-dsl/term/prefix/), and [`regexp`]({{site.url}}{{site.baseurl}}/query-dsl/term/regexp/) queries perform better on `wildcard` fields.
+{: .tip}
+
+## Example
+
+Create a mapping with a `wildcard` field:
+
+```json
+PUT movies
```
Comment (msfroh) [inline _field-types/supported-field-types/wildcard.md:28]: Oh... I changed the field name from the example that I copied to highlight that wildcard fields are good for log lines, but I forgot to change the name of the index.

Comment (kolchfa-aws) [inline _field-types/supported-field-types/wildcard.md:28]:
```suggestion
PUT logs
```

Comment (kolchfa-aws) [reply _field-types/supported-field-types/wildcard.md]: @msfroh Changed the index name to "logs"

**Review by natebower — APPROVED** [review]
LGTM

**Pull request merged**